client/services/code_preprocessors/PythonCodePreprocessorService.js

```diff
@@ -22,7 +22,37 @@ tie.factory('PythonCodePreprocessorService', [
   function(WRAPPER_CLASS_NAME, VARNAME_TEST_RESULTS) {
     var START_INDENT = '    ';
 
+    // Wraps a WRAPPER_CLASS_NAME class around the series of functions in a
+    // given code snippet.
+    var wrapCodeIntoClass = function(code) {
+      var codeLines = code.trim().split('\n');
+      var firstLine = 'import time\n\n\nclass ' + WRAPPER_CLASS_NAME + '(object):';
+      var systemFunctions = [
+        START_INDENT + 'def extendString(self, s, length):',
+        START_INDENT + '    return s * length'
+      ];
+      var subsequentLines = codeLines.map(function(line) {
+        if (line.indexOf('def') === 0) {
+          var leftParenIndex = line.indexOf('(');
+          if (leftParenIndex === -1) {
+            throw Error('Incomplete line: missing "(" in def statement.');
+          }
+          return (
+            START_INDENT +
+            line.slice(0, leftParenIndex + 1) +
+            'self, ' +
+            line.slice(leftParenIndex + 1));
+        } else {
+          return START_INDENT + line;
+        }
+      });
+      var newCodeLines = [firstLine].concat(systemFunctions);
+      newCodeLines = newCodeLines.concat(subsequentLines);
+      return newCodeLines.join('\n');
+    };
+
     var jsonVariableToPython = function(jsonVariable) {
+      console.log('var = ' + jsonVariable);
```

Comment:
Right -- sorry, only left these in so that we could try to fix the testing thing. Normally before a PR I'd remove these.